src/libstd/keyword_docs.rs

```diff
@@ -286,11 +286,13 @@ mod fn_keyword { }
 //
 /// The `for` keyword.
 ///
-/// `for` is primarily used in for-in-loops, but it has a few other pieces of syntactic uses such as
-/// `impl Trait for Type` (see [`impl`] for more info on that). for-in-loops, or to be more
-/// precise, iterator loops, are a simple syntactic sugar over an exceedingly common practice
-/// within Rust, which is to loop over an iterator until that iterator returns `None` (or `break`
-/// is called).
+/// `for` is primarily used in for-in-loops, but it has a few other pieces of syntactic
```

Comment:
```suggestion
/// `for` is used in for-in-loops, but it is used in other syntactic locations as well.
```
I don't think the assertion of "primarily" is necessarily true actually; `impl Trait for Type` is probably quite common as well.